### ISSUE TITLE
Unmangle Rust function names?

### ISSUE THREAD

**Issue by bnjbvr** [wolfpld/tracy#279]
It would be cool to unmangle rust function names when collecting (or alternatively later, displaying them the first time), using e.g. https://github.com/luser/rustfilt or something similar.

**Comment by Geod24** [comment]
In general, the ability to plug in a demangler would be welcome. The same problematic applies to D.

**Comment by wolfpld** [comment]
With e3682668 you can now provide your own implementation of demangler:

```c++
extern "C" int ___tracy_demangle( const char* mangled, char* out, size_t len );

#ifndef TRACY_DEMANGLE
extern "C" int ___tracy_demangle( const char* mangled, char* out, size_t len )
{
    if( !mangled || mangled[0] != '_' ) return 0;
    int status;
    abi::__cxa_demangle( mangled, out, &len, &status );
    return status == 0;
}
#endif
```

**Comment by omerfirmak** [comment]
`__cxa_demangle` requires the buffer passed to it to be heap allocated, but if I am not mistaken you are allocating it on the stack. @wolfpld

**Comment by wolfpld** [comment]
Yes, this is a known problem with the available external API. Tracy is not expected to be making any visible allocations (as you may want to track any such allocation with Tracy (and there are use cases in which people have overriden `malloc` and `free` for such purpose)), so it uses its own separate allocator to workaround this problem. This would not work well with an external library requiring a specific heap (provided by libc in this case) to be used within the profiler.

The provided buffer currently has 64 KB, which should be enough for any forseeable application.

**Comment by wolfpld** [comment]
> The provided buffer currently has 64 KB, which should be enough for any forseeable application.

Of course it turned out to be not enough. This was fixed in #430.
